codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,295 @@
+            'each on a separate line. If not specified, the '
+            'the worker will fail to start',
+        )
+        subparser.add_argument(
+            '--slurm-work-dir',
+            default='slurm-worker-scratch',
+            help='Directory where to store Slurm batch scripts, logs, etc.',
+        )
+
+    def __init__(self, args):
+        super().__init__(args)
+        self.username = self.args.user
+        # A set of newly submitted job id to keep tracking worker status, as worker might not be created right away.
+        self.submitted_jobs = self.load_worker_jobs()
+
+    def load_worker_jobs(self):
+        """
+        Load worker jobs that are created using SlurmWorkerManager and
+        owned by the current user from the Slurm scheduling queue.
+        :return: a set of job id
+        """
+        # Get all the Slurm workers that are submitted by SlurmWorkerManager and owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID:NAME (header won't be included with "--noheader" option)
+        # 1487896,john-job-3157358
+        # 1478830,john-job-5234492
+        submitted_jobs = set()
+        jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '--format', '%A,%j', '--noheader']
+        )
+        for job in jobs.strip().split():
+            job_id, job_name = job.split(',')
+            if job_name.startswith(self.username) and self.args.job_name in job_name:
+                submitted_jobs.add(job_id)
+        return submitted_jobs
+
+    def get_worker_jobs(self):
+        """
+        Return a list of workers in RUNNING and PENDING state.
+        The current Slurm version on NLP cluster is 17.11.13-2, which doesn't have rest api provided.
+        """
+        # Documentation can be found at https://slurm.schedmd.com
+        # Since allocating resource for a worker may take a while, we periodically check
+        # for worker status and remove those workers that failed at starting phase.
+        jobs_to_remove = set()
+        for job_id in self.submitted_jobs:
+            job_acct = self.run_command([self.SACCT, '-j', job_id, '--format', 'state'])
+            if 'FAILED' in job_acct:
+                jobs_to_remove.add(job_id)
+                logger.error("Failed to start job {}".format(job_id))
+            elif 'COMPLETING' in job_acct or 'COMPLETED' in job_acct or 'CANCELLED' in job_acct:
+                jobs_to_remove.add(job_id)
+        self.submitted_jobs = self.submitted_jobs - jobs_to_remove
+        logger.info("Submitted jobs: ".format(self.submitted_jobs))
+
+        # Get all the Slurm workers that are submitted by SlurmWorkerManager and owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID:STATE (header won't be included with "--noheader" option)
+        # 1478828,PENDING
+        # 1478830,PENDING
+        jobs = self.run_command(
```

Comment:
You don't want to count all of the things returned by `squeue -u`, right? I think you want to match the name prefix, like how it's done in `load_worker_jobs`.

Reply:
Oh, I see that you later filter by checking if it's in `self.submitted_jobs`. let me know if i'm wrong about that.

Reply:
That's correct. It will only report jobs in `self.submitted_jobs` which has a filter on the name prefix.